lib/index.js

```diff
@@ -192,6 +192,42 @@ module.exports = function(session) {
 		});
 	};
 
+	MySQLStore.prototype.touch = function(session_id, data, cb) {
+
+		debug_log('Touching session: ' + session_id);
+
+		var expires;
+
+		if (data.cookie && data.cookie.expires) {
+			expires = data.cookie.expires;
+		} else {
+			expires = new Date(Date.now() + this.options.expiration);
+		}
```

Comment:
When "data.cookie.expires" expires, session in user's browser would be deleted. We should not store it in mysql anymore. So i did not set new value to "expires" if "data.cookie.expires" is set.

PS: Excuse my poor English )))